src/ControllerPose.cpp

```diff
@@ -1,17 +1,30 @@
 #include "ControllerPose.h"
 
-#include <queue>
-#include <utility>
-
 #include "DriverLog.h"
 #include "Quaternion.h"
 
-ControllerPose::ControllerPose(vr::ETrackedControllerRole shadowDeviceOfRole,
-                               std::string thisDeviceManufacturer,
-                               VRPoseConfiguration_t poseConfiguration)
-    : m_shadowDeviceOfRole(shadowDeviceOfRole),
-      m_thisDeviceManufacturer(std::move(thisDeviceManufacturer)),
-      m_poseConfiguration(poseConfiguration) {
+struct CalibrationDataIn {
+  uint8_t start;
+};
+
+ControllerPose::ControllerPose(vr::ETrackedControllerRole shadowDeviceOfRole, std::string thisDeviceManufacturer, VRPoseConfiguration_t poseConfiguration)
+    : m_shadowDeviceOfRole(shadowDeviceOfRole), m_thisDeviceManufacturer(std::move(thisDeviceManufacturer)), m_poseConfiguration(poseConfiguration) {
+  m_calibrationPipe =
+      std::make_unique<NamedPipeUtil>("\\\\.\\pipe\\vrapplication\\functions\\autocalibrate\\" +
+                                          std::string(shadowDeviceOfRole == vr::ETrackedControllerRole::TrackedControllerRole_RightHand ? "right" : "left"),
+                                      sizeof(CalibrationDataIn));
+
+  m_calibrationPipe->Start([&](LPVOID data) {
+    CalibrationDataIn *calibrationData = (CalibrationDataIn *)data;
+
+    if (calibrationData->start) {
+      DriverLog("Starting calibration via external application");
+      StartCalibration();
```

Comment:
Should we check if it's currently calibrating?

Reply:
I think it should be fine, worse thing that happens if its called twice the pose updates, which could be intended by the user (ie you can start calibrating multiple times before stopping)